generativeai/src/main/java/com/google/ai/client/generativeai/Chat.kt

```diff
@@ -156,4 +162,13 @@ class Chat(private val model: GenerativeModel, val history: MutableList<Content>
       throw InvalidStateException("Chat prompts should come from the 'user' role.")
     }
   }
+
+  private fun attemptLock() {
+    if (!lock.tryAcquire()) {
+      throw IllegalStateException(
```

Comment:
Should be using our own variant, which is a subclass of `GoogleGenerativeAIException`

```suggestion
      throw InvalidStateException(
```